web/app.js

```diff
@@ -817,7 +817,10 @@ const PDFViewerApplication = {
 
         return loadingErrorMessage.then(msg => {
           this.error(msg, { message });
-          throw new Error(msg);
+
+          exception.message = msg;
+
+          throw exception;
```

Comment:
This is still modifying, albeit less than the previously, the original exception though.

Why not simply `throw exception;` here, since the translation is mostly intended for the default viewer `errorWrapper`?

Reply:
Thanks for your feedback, @Snuffleupagus!

That would of course also work. However it would be a BC break if an application using pdf.js relies on the translated exception message (e.g. to display user-facing error messages).

Reply:
The default viewer was created for usage in Firefox, and is *not* part of the versioned [public API](see https://github.com/mozilla/pdf.js/blob/master/src/display/api.js).
There's also the "viewer components", see https://github.com/mozilla/pdf.js/blob/master/web/pdf_viewer.component.js and the examples in https://github.com/mozilla/pdf.js/tree/master/examples/components, which we also attempt to not break unnecessarily.

For the [default viewer](https://github.com/mozilla/pdf.js/blob/master/web/app.js), while we don't go out of our way to break backwards compatibility, there's absolutely *no guarantees* that there won't be changes; note also that we even ask people not to use it as -is e.g. in http://mozilla.github.io/pdf.js/getting_started/#introduction (emphasis mine):
> The viewer is built on the display layer and is the UI for PDF viewer in Firefox and the other browser extensions within the project. It can be a good starting point for building your own viewer. ***However, we do ask if you plan to embed the viewer in your own site, that it not just be an unmodified version. Please re-skin it or build upon it.***

---

> [...] if an application using pdf.js relies on the translated exception message (e.g. to display user-facing error messages).

Apart from the above, this particular situation also seem to be enough of an edge-case that I wouldn't really consider the change I suggested above to be a problem.
It'd even be more consistent, since it means that the viewer is now returning *exactly* the same Errors as if the "raw" API was used directly.

Reply:
I agree with the above. This looks good if the original error is re-thrown, i.e., unchanged.

Reply:
That sounds good, thank you very much for the feedback. I've adapted the code accordingly.